Quadriga/src/main/java/edu/asu/spring/quadriga/dao/workbench/impl/ProjectAccessDAO.java

```diff
@@ -12,90 +12,88 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getNrOfOwnedProjects(String userName) {
+        Query query = sessionFactory.getCurrentSession()
+                .createQuery("SELECT count(p.projectid) FROM ProjectDTO p WHERE p.projectowner.username =:userName");
+        query.setParameter("userName", userName);
+        return ((Number) query.iterate().next()).intValue();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getNrOfProjectsCollaboratingOn(String userName, String collaboratorRole) {
+        Query query = sessionFactory.getCurrentSession().createQuery(
+                "SELECT count(pc.collaboratorDTOPK.projectid) FROM ProjectCollaboratorDTO pc WHERE pc.collaboratorDTOPK.collaboratoruser =:userName AND pc.collaboratorDTOPK.collaboratorrole =:collaboratorRole");
+        query.setParameter("userName", userName);
+        query.setParameter("collaboratorRole", collaboratorRole);
+        return ((Number) query.iterate().next()).intValue();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isCollaborator(String userName, String collaboratorRole, String projectId) {
+        Query query = sessionFactory.getCurrentSession().createQuery(
+                "SELECT count(pc.collaboratorDTOPK.projectid) FROM ProjectCollaboratorDTO pc WHERE pc.collaboratorDTOPK.collaboratoruser =:userName AND pc.collaboratorDTOPK.collaboratorrole =:collaboratorRole AND pc.collaboratorDTOPK.projectid =:projectId");
+        query.setParameter("userName", userName);
+        query.setParameter("collaboratorRole", collaboratorRole);
+        query.setParameter("projectId", projectId);
+        return ((Number) query.iterate().next()).intValue() > 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getProjectIdByUnixName(String unixName) {
+        Query query = sessionFactory.getCurrentSession().getNamedQuery("ProjectDTO.findByUnixname");
+        query.setString("unixname", unixName);
+
+        ProjectDTO project = (ProjectDTO) query.uniqueResult();
+
+        if (project != null) {
+            return project.getProjectid();
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int isUserEditorOfProject(String userName, String collaboratorRole, String projectId) {
+        Query query = sessionFactory.getCurrentSession().createQuery(
+                "SELECT count(pc.collaboratorDTOPK.projectid) FROM ProjectCollaboratorDTO pc WHERE pc.collaboratorDTOPK.collaboratoruser =:userName AND pc.collaboratorDTOPK.collaboratorrole =:collaboratorRole AND pc.collaboratorDTOPK.projectid =:projectId");
+        query.setParameter("userName", userName);
+        query.setParameter("collaboratorRole", collaboratorRole);
+        query.setParameter("projectId", projectId);
+
+        return ((Number) query.uniqueResult()).intValue();
+    }
```

Comment:
Only indentation is changed in these files. 'IProjectAccessDAO" and "ProjectAccessDAO".